crates/miden-objects/src/address/routing_parameters.rs

```diff
@@ -242,6 +257,150 @@ impl Deserializable for RoutingParameters {
+        None
+    } else {
+        Some(tag_len)
+    };
+
+    let addr_interface = receiver_profile & 0b0000_0111_1111_1111;
+    let addr_interface = AddressInterface::try_from(addr_interface).map_err(|err| {
+        AddressError::decode_error_with_source("failed to decode address interface", err)
+    })?;
+
+    Ok((addr_interface, note_tag_len))
+}
+
+/// Append encryption key variant discriminant and key to the provided vector of bytes.
+fn encode_encryption_key(key: &SealingKey, encoded: &mut Vec<u8>) {
+    match key {
+        SealingKey::X25519XChaCha20Poly1305(pk) => {
+            encoded.push(ENCRYPTION_KEY_X25519_XCHACHA20POLY1305);
+            encoded.extend(&pk.to_bytes());
+        },
+        SealingKey::K256XChaCha20Poly1305(pk) => {
+            encoded.push(ENCRYPTION_KEY_K256_XCHACHA20POLY1305);
+            encoded.extend(&pk.to_bytes());
+        },
+        SealingKey::X25519AeadRpo(pk) => {
+            encoded.push(ENCRYPTION_KEY_X25519_AEAD_RPO);
+            encoded.extend(&pk.to_bytes());
+        },
+        SealingKey::K256AeadRpo(pk) => {
+            encoded.push(ENCRYPTION_KEY_K256_AEAD_RPO);
+            encoded.extend(&pk.to_bytes());
+        },
+    }
+}
+
+/// Reads the encryption key from the provided bytes.
+fn decode_encryption_key(
+    byte_iter: &mut impl ExactSizeIterator<Item = u8>,
+) -> Result<SealingKey, AddressError> {
+    // Need at least 1 byte for discriminant
+    if byte_iter.len() < 1 {
+        return Err(AddressError::decode_error(
+            "expected at least 1 byte for encryption key variant",
+        ));
+    };
+
+    // Read variant discriminant
+    let variant = byte_iter.next().expect("variant byte should exist");
+
+    // Reconstruct the appropriate PublicEncryptionKey variant
+    let public_encryption_key = match variant {
+        ENCRYPTION_KEY_X25519_XCHACHA20POLY1305 => {
+            SealingKey::X25519XChaCha20Poly1305(read_x25519_pub_key(byte_iter)?)
+        },
+        ENCRYPTION_KEY_K256_XCHACHA20POLY1305 => {
+            SealingKey::K256XChaCha20Poly1305(read_k256_pub_key(byte_iter)?)
+        },
+        ENCRYPTION_KEY_X25519_AEAD_RPO => {
+            SealingKey::X25519AeadRpo(read_x25519_pub_key(byte_iter)?)
+        },
+        ENCRYPTION_KEY_K256_AEAD_RPO => SealingKey::K256AeadRpo(read_k256_pub_key(byte_iter)?),
+        other => {
+            return Err(AddressError::decode_error(format!(
+                "unknown encryption key variant: {}",
+                other
+            )));
+        },
+    };
+
+    Ok(public_encryption_key)
+}
+
+fn read_x25519_pub_key(
+    byte_iter: &mut impl ExactSizeIterator<Item = u8>,
+) -> Result<eddsa_25519::PublicKey, AddressError> {
+    if byte_iter.len() < ED25519_PUBLIC_KEY_LENGTH {
+        return Err(AddressError::decode_error(format!(
+            "expected {} bytes to decode Ed25519 public key",
+            ED25519_PUBLIC_KEY_LENGTH
+        )));
```

Comment:
Nit: Since we publicly (e.g the `SealingKey` variants) only refer to `X25519` variants, it may be clearer to use `X25519` in the constant names and in the error message here as well, even if it's technically correct to write `Ed25519`.